<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>biotaphy.analyses.helpers package &mdash; Biotaphy BiotaPhyPy 2.0.0a1.post0.dev8 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/graphviz.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> Biotaphy BiotaPhyPy
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../pages/installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../analyses/analyses.html">Analyses</a></li>
<li class="toctree-l1"><a class="reference internal" href="../pages/data_formats.html">Data Formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="../pages/references.html">References</a></li>
<li class="toctree-l1"><a class="reference internal" href="../pages/contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../pages/about.html">About</a></li>
<li class="toctree-l1"><a class="reference internal" href="../autoapi/index.html">API Reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Biotaphy BiotaPhyPy</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>biotaphy.analyses.helpers package</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/source/biotaphy.analyses.helpers.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="biotaphy-analyses-helpers-package">
<h1>biotaphy.analyses.helpers package<a class="headerlink" href="#biotaphy-analyses-helpers-package" title="Permalink to this headline"></a></h1>
<section id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Permalink to this headline"></a></h2>
</section>
<section id="module-biotaphy.analyses.helpers.data_readers">
<span id="biotaphy-analyses-helpers-data-readers-module"></span><h2>biotaphy.analyses.helpers.data_readers module<a class="headerlink" href="#module-biotaphy.analyses.helpers.data_readers" title="Permalink to this headline"></a></h2>
<p>Module containing tools for reading alignment files in various formats.</p>
<dl class="py exception">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.data_readers.AlignmentIOError">
<em class="property"><span class="pre">exception</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">biotaphy.analyses.helpers.data_readers.</span></span><span class="sig-name descname"><span class="pre">AlignmentIOError</span></span><a class="reference internal" href="../_modules/biotaphy/analyses/helpers/data_readers.html#AlignmentIOError"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#biotaphy.analyses.helpers.data_readers.AlignmentIOError" title="Permalink to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Exception</span></code></p>
<p>Wrapper class for alignment errors.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.data_readers.create_sequence_list_from_dict">
<span class="sig-prename descclassname"><span class="pre">biotaphy.analyses.helpers.data_readers.</span></span><span class="sig-name descname"><span class="pre">create_sequence_list_from_dict</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">values_dict</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/biotaphy/analyses/helpers/data_readers.html#create_sequence_list_from_dict"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#biotaphy.analyses.helpers.data_readers.create_sequence_list_from_dict" title="Permalink to this definition"></a></dt>
<dd><p>Creates a list of sequences from a dictionary.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>values_dict</strong> (<em>dict</em>) – A dictionary of taxon name keys and a list of
values for each value.</p>
</dd>
</dl>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul>
<li><p>The dictionary should have structure:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="s2">&quot;</span><span class="si">{taxon_name}</span><span class="s2">&quot;</span> <span class="p">:</span> <span class="p">[{</span><span class="n">values</span><span class="p">}]</span>
<span class="p">}</span>
</pre></div>
</div>
</li>
</ul>
</div>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>A list of Sequence objects and None for headers.</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>list</p>
</dd>
<dt class="field-odd">Raises</dt>
<dd class="field-odd"><p><a class="reference internal" href="#biotaphy.analyses.helpers.data_readers.AlignmentIOError" title="biotaphy.analyses.helpers.data_readers.AlignmentIOError"><strong>AlignmentIOError</strong></a> – If a dictionary value is not a list.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.data_readers.get_character_matrix_from_sequences_list">
<span class="sig-prename descclassname"><span class="pre">biotaphy.analyses.helpers.data_readers.</span></span><span class="sig-name descname"><span class="pre">get_character_matrix_from_sequences_list</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sequences</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">var_headers</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/biotaphy/analyses/helpers/data_readers.html#get_character_matrix_from_sequences_list"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#biotaphy.analyses.helpers.data_readers.get_character_matrix_from_sequences_list" title="Permalink to this definition"></a></dt>
<dd><p>Converts a list of sequences into a character matrix.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>sequences</strong> (<code class="xref py py-obj docutils literal notranslate"><span class="pre">list</span></code> of <code class="xref py py-obj docutils literal notranslate"><span class="pre">Sequence</span></code>) – A list of Sequence objects
to be converted.</p></li>
<li><p><strong>var_headers</strong> (<code class="xref py py-obj docutils literal notranslate"><span class="pre">list</span></code> of headers, optional) – If provided, uses these
as variable headers for the columns in the matrix.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A matrix of sequence data.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Matrix</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.data_readers.load_alignment_from_filename">
<span class="sig-prename descclassname"><span class="pre">biotaphy.analyses.helpers.data_readers.</span></span><span class="sig-name descname"><span class="pre">load_alignment_from_filename</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">filename</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/biotaphy/analyses/helpers/data_readers.html#load_alignment_from_filename"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#biotaphy.analyses.helpers.data_readers.load_alignment_from_filename" title="Permalink to this definition"></a></dt>
<dd><p>Attempts to load an alignment from a file path by guessing schema.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>filename</strong> (<em>str</em>) – The file location containing the alignment</p>
</dd>
<dt class="field-even">Raises</dt>
<dd class="field-even"><p><strong>RuntimeError</strong> – Raised with the method needed to load the alignment
    cannot be determined.</p>
</dd>
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>Containing a list of sequences and headers</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>tuple</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.data_readers.read_csv_alignment_flo">
<span class="sig-prename descclassname"><span class="pre">biotaphy.analyses.helpers.data_readers.</span></span><span class="sig-name descname"><span class="pre">read_csv_alignment_flo</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">csv_flo</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/biotaphy/analyses/helpers/data_readers.html#read_csv_alignment_flo"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#biotaphy.analyses.helpers.data_readers.read_csv_alignment_flo" title="Permalink to this definition"></a></dt>
<dd><p>Reads a CSV file-like object and return a list of sequences and headers.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>csv_flo</strong> (<em>file-like</em>) – A file-like object with CSV alignment data.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A list of Sequence objects and headers.</p>
</dd>
<dt class="field-odd">Raises</dt>
<dd class="field-odd"><p><a class="reference internal" href="#biotaphy.analyses.helpers.data_readers.AlignmentIOError" title="biotaphy.analyses.helpers.data_readers.AlignmentIOError"><strong>AlignmentIOError</strong></a> – If the number of columns is inconsistent across the
    sequences.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.data_readers.read_json_alignment_flo">
<span class="sig-prename descclassname"><span class="pre">biotaphy.analyses.helpers.data_readers.</span></span><span class="sig-name descname"><span class="pre">read_json_alignment_flo</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">json_flo</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/biotaphy/analyses/helpers/data_readers.html#read_json_alignment_flo"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#biotaphy.analyses.helpers.data_readers.read_json_alignment_flo" title="Permalink to this definition"></a></dt>
<dd><p>Read a JSON file-like object and return a list of sequences and headers.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>json_flo</strong> (<em>file-like</em>) – A file-like object with JSON alignment data.</p>
</dd>
</dl>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul>
<li><p>File should have structure:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="s2">&quot;headers&quot;</span> <span class="p">:</span> <span class="p">[{</span><span class="n">header_names</span><span class="p">}],</span>
    <span class="s2">&quot;values&quot;</span> <span class="p">:</span> <span class="p">[</span>
        <span class="p">{</span>
            <span class="s2">&quot;name&quot;</span> <span class="p">:</span> <span class="s2">&quot;</span><span class="si">{taxon_name}</span><span class="s2">&quot;</span><span class="p">,</span>
            <span class="s2">&quot;values&quot;</span> <span class="p">:</span> <span class="p">[{</span><span class="n">values</span><span class="p">}]</span>
        <span class="p">}</span>
    <span class="p">]</span>
<span class="p">}</span>
</pre></div>
</div>
</li>
</ul>
</div>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>A list of Sequence objects and headers.</p>
</dd>
<dt class="field-even">Raises</dt>
<dd class="field-even"><p><a class="reference internal" href="#biotaphy.analyses.helpers.data_readers.AlignmentIOError" title="biotaphy.analyses.helpers.data_readers.AlignmentIOError"><strong>AlignmentIOError</strong></a> – If headers are provided but they are not a list.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.data_readers.read_phylip_alignment_flo">
<span class="sig-prename descclassname"><span class="pre">biotaphy.analyses.helpers.data_readers.</span></span><span class="sig-name descname"><span class="pre">read_phylip_alignment_flo</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">phylip_flo</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/biotaphy/analyses/helpers/data_readers.html#read_phylip_alignment_flo"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#biotaphy.analyses.helpers.data_readers.read_phylip_alignment_flo" title="Permalink to this definition"></a></dt>
<dd><p>Reads a phylip alignment file-like object and return the sequences.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>phylip_flo</strong> (<em>file-like</em>) – The phylip file-like object.</p>
</dd>
</dl>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul class="simple">
<li><dl class="simple">
<dt>We assume that the phylip files are extended and not strict (in terms</dt><dd><p>of how many characters for taxon names).</p>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>The phylip file is in the format::</dt><dd><p>numoftaxa numofsites
seqlabel sequence
seqlabel sequence</p>
</dd>
</dl>
</li>
</ul>
</div>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>A list of Sequence objects.</p>
</dd>
<dt class="field-even">Raises</dt>
<dd class="field-even"><p><a class="reference internal" href="#biotaphy.analyses.helpers.data_readers.AlignmentIOError" title="biotaphy.analyses.helpers.data_readers.AlignmentIOError"><strong>AlignmentIOError</strong></a> – If there is a problem creating sequences.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.data_readers.read_table_alignment_flo">
<span class="sig-prename descclassname"><span class="pre">biotaphy.analyses.helpers.data_readers.</span></span><span class="sig-name descname"><span class="pre">read_table_alignment_flo</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">table_flo</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/biotaphy/analyses/helpers/data_readers.html#read_table_alignment_flo"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#biotaphy.analyses.helpers.data_readers.read_table_alignment_flo" title="Permalink to this definition"></a></dt>
<dd><p>Reads a table from a file-like object.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>table_flo</strong> (<em>file-like</em>) – A file-like object containing table data.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A list of Sequence objects.</p>
</dd>
<dt class="field-odd">Raises</dt>
<dd class="field-odd"><p><a class="reference internal" href="#biotaphy.analyses.helpers.data_readers.AlignmentIOError" title="biotaphy.analyses.helpers.data_readers.AlignmentIOError"><strong>AlignmentIOError</strong></a> – If there is a problem creating sequences.</p>
</dd>
</dl>
</dd></dl>

</section>
<section id="biotaphy-analyses-helpers-permutation-testing-module">
<h2>biotaphy.analyses.helpers.permutation_testing module<a class="headerlink" href="#biotaphy-analyses-helpers-permutation-testing-module" title="Permalink to this headline"></a></h2>
</section>
<section id="module-biotaphy.analyses.helpers.sequence">
<span id="biotaphy-analyses-helpers-sequence-module"></span><h2>biotaphy.analyses.helpers.sequence module<a class="headerlink" href="#module-biotaphy.analyses.helpers.sequence" title="Permalink to this headline"></a></h2>
<p>Module containing sequence class.</p>
<dl class="py class">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.sequence.Sequence">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">biotaphy.analyses.helpers.sequence.</span></span><span class="sig-name descname"><span class="pre">Sequence</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">''</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seq</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">''</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/biotaphy/analyses/helpers/sequence.html#Sequence"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#biotaphy.analyses.helpers.sequence.Sequence" title="Permalink to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Barebones class for sequences.</p>
<p>This is a barebones class for sequences.  These can be aligned or not and
can be any type of alphabet.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.sequence.Sequence.name">
<span class="sig-name descname"><span class="pre">name</span></span><a class="headerlink" href="#biotaphy.analyses.helpers.sequence.Sequence.name" title="Permalink to this definition"></a></dt>
<dd><p>A name for this sequence.</p>
<dl class="field-list simple">
<dt class="field-odd">Type</dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.sequence.Sequence.seq">
<span class="sig-name descname"><span class="pre">seq</span></span><a class="headerlink" href="#biotaphy.analyses.helpers.sequence.Sequence.seq" title="Permalink to this definition"></a></dt>
<dd><p>A sequence string.</p>
<dl class="field-list simple">
<dt class="field-odd">Type</dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.sequence.Sequence.qualstr">
<span class="sig-name descname"><span class="pre">qualstr</span></span><a class="headerlink" href="#biotaphy.analyses.helpers.sequence.Sequence.qualstr" title="Permalink to this definition"></a></dt>
<dd><p>A string of sequence characters.</p>
<dl class="field-list simple">
<dt class="field-odd">Type</dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.sequence.Sequence.qualarr">
<span class="sig-name descname"><span class="pre">qualarr</span></span><a class="headerlink" href="#biotaphy.analyses.helpers.sequence.Sequence.qualarr" title="Permalink to this definition"></a></dt>
<dd><p>A list of offset code points.</p>
<dl class="field-list simple">
<dt class="field-odd">Type</dt>
<dd class="field-odd"><p>list</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.sequence.Sequence.cont_values">
<span class="sig-name descname"><span class="pre">cont_values</span></span><a class="headerlink" href="#biotaphy.analyses.helpers.sequence.Sequence.cont_values" title="Permalink to this definition"></a></dt>
<dd><p>A list of continuous values for a sequence.</p>
<dl class="field-list simple">
<dt class="field-odd">Type</dt>
<dd class="field-odd"><p>list</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.sequence.Sequence.get_fasta">
<span class="sig-name descname"><span class="pre">get_fasta</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/biotaphy/analyses/helpers/sequence.html#Sequence.get_fasta"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#biotaphy.analyses.helpers.sequence.Sequence.get_fasta" title="Permalink to this definition"></a></dt>
<dd><p>Get a fasta string.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>str - A fasta string for the sequence.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.sequence.Sequence.get_fastq">
<span class="sig-name descname"><span class="pre">get_fastq</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/biotaphy/analyses/helpers/sequence.html#Sequence.get_fastq"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#biotaphy.analyses.helpers.sequence.Sequence.get_fastq" title="Permalink to this definition"></a></dt>
<dd><p>Get a fastq string.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>str - A fastq string for the sequence.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.sequence.Sequence.set_cont_values">
<span class="sig-name descname"><span class="pre">set_cont_values</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">values</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/biotaphy/analyses/helpers/sequence.html#Sequence.set_cont_values"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#biotaphy.analyses.helpers.sequence.Sequence.set_cont_values" title="Permalink to this definition"></a></dt>
<dd><p>Set the continuous values for the sequence.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>values</strong> (<em>list</em>) – A list of values.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.sequence.Sequence.set_qualarr">
<span class="sig-name descname"><span class="pre">set_qualarr</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">qual</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/biotaphy/analyses/helpers/sequence.html#Sequence.set_qualarr"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#biotaphy.analyses.helpers.sequence.Sequence.set_qualarr" title="Permalink to this definition"></a></dt>
<dd><p>Set the qualarr attribute.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>qual</strong> (<code class="xref py py-obj docutils literal notranslate"><span class="pre">list</span></code> of <code class="xref py py-obj docutils literal notranslate"><span class="pre">int</span></code>) – A list of code point integers.</p>
</dd>
</dl>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul class="simple">
<li><p>An offset of 33 is assumed.</p></li>
</ul>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="biotaphy.analyses.helpers.sequence.Sequence.set_qualstr">
<span class="sig-name descname"><span class="pre">set_qualstr</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">qual</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/biotaphy/analyses/helpers/sequence.html#Sequence.set_qualstr"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#biotaphy.analyses.helpers.sequence.Sequence.set_qualstr" title="Permalink to this definition"></a></dt>
<dd><p>Set the qualstr attribute.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>qual</strong> (<em>str</em>) – A new string to use for qualstr.</p>
</dd>
</dl>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul class="simple">
<li><p>An offset of 33 is assumed.</p></li>
</ul>
</div>
</dd></dl>

</dd></dl>

</section>
<section id="module-biotaphy.analyses.helpers">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-biotaphy.analyses.helpers" title="Permalink to this headline"></a></h2>
<p>Helpers module __init__.</p>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Biotaphy Team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>